### ISSUE TITLE
[Bug]: Empty `pr_branch` output variable

### ISSUE THREAD

**Issue by kdeldycke** [AndreasAugustin/actions-template-sync#476]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Describe the bug

It looks like the new `pr_branch` variable output (implemented by #472 / #473) is not set if there is already a PR previously created by `actions-template-sync`.

### Expected Behavior

I expect the `pr_branch` variable to always be set to the active / unmerged PR that is tracked an maintained by `actions-template-sync`.

### Current Behavior

I have a workflow that essentially looks like this:
```yaml
(...)

jobs:

  awesome-template-sync:
    runs-on: ubuntu-22.04
    steps:
      - uses: actions/checkout@v4.1.1

      - name: Sync from template repo
        id: template_sync
        uses: AndreasAugustin/actions-template-sync@v1.8.0
        with:
          # We need custom PAT with workflows permissions so we can update all the boilerplate .github files from
          # awesome-template.
          github_token: ${{ secrets.WORKFLOW_UPDATE_GITHUB_PAT || secrets.GITHUB_TOKEN }}
          source_repo_path: kdeldycke/awesome-template
          pr_title: "[sync] Updates from `awesome-template` repository"
          pr_commit_msg: "[sync] Updates from `awesome-template` repository"
          pr_branch_name_prefix: "sync-awesome-template"
          pr_labels: "📚 documentation"
          is_pr_cleanup: true

      - name: Print PR branch ID
        run: |
          echo "PR branch ID: ${{ steps.template_sync.outputs.pr_branch }}"

      (...)
```

The full workflow code is available at: https://github.com/kdeldycke/awesome-billing/blob/b4686574a1a328f001ca1410f65b6137f990763d/.github/workflows/docs.yaml#L16-L59

Its execution log is available at: https://github.com/kdeldycke/awesome-billing/actions/runs/8113301740/job/22199139632

On execution, you can find that the step printing the `pr_branch` variable returns nothing:

![Screenshot 2024-03-02 at 11 14 28](https://github.com/AndreasAugustin/actions-template-sync/assets/159718/e314ec69-c522-4991-9df3-95e26d99d52e)

This is probably an edge-case because there is already an unmerged (and conflicting) PR that was previously created by `actions-template-sync`:

![Screenshot 2024-03-02 at 11 15 05](https://github.com/AndreasAugustin/actions-template-sync/assets/159718/ab11e686-3f38-45e8-983e-99d294ab2b6b)

I expect here `actions-template-sync` should identify the already existing `sync-awesome-template_d48d756` PR and branch, update it, and return its ID in `pr_branch` variable.

Note that the behavior is the same wether `is_pr_cleanup` parameter is `true` or `false`.

### Additional Information/Context

This was encounter while playing with 1.8.0 to implement a temporary workaround for #467.

### Template sync version Version

1.8.0

**Comment by AndreasAugustin** [comment]
@kdeldycke thanks for the report. Indeed I forgot to handle some 'edge-cases' (thanks for the hint within the description).
Sorry for that :ninja: 

The edge-case in your case is the following:

- The action is checking if the branch is already available remotely (the prefix is the commit short ID) and exiting.
- The output definition is with `v1.8.0` at the end of the action and not done for exit cases

Your action log part:

```txt
::info::Missing env variable 'UPSTREAM_BRANCH' setting to remote default main
::info::current git hash: b4686574a1a328f001ca1410f65b6137f990763d
Check new changes
  a5d4c1b6d368ba257[28](https://github.com/kdeldycke/awesome-billing/actions/runs/8113301740/job/22199139632#step:4:30)94b0e93ac16cd65a6f02e	refs/heads/sync-awesome-template_d48d756
  ::warn::Git branch 'sync-awesome-template_d48d756' exists in the remote repository
```

related code part [src/sync_template.sh](src/sync_template.sh) line 51

```bash
function check_branch_remote_existing() {
  git ls-remote --exit-code --heads origin "${NEW_BRANCH}" || BRANCH_DOES_NOT_EXIST=true

  if [[ "${BRANCH_DOES_NOT_EXIST}" != true ]]; then
    warn "Git branch '${NEW_BRANCH}' exists in the remote repository"
# following line will be added with v1.8.1    
# set_github_action_outputs
    exit 0
  fi
}

```

With `v1.8.1`  Your case should be fixed.

**Comment by kdeldycke** [comment]
Wow, thanks @AndreasAugustin for the incredibly fast investigation and fix! I can confirm v1.8.1 fix the issue and `pr_branch` is properly set.

You can check the proof here: https://github.com/kdeldycke/awesome-billing/actions/runs/8122438292/job/22201872306#step:5:5
